src/enzo/StarParticleRadTransfer.C

```diff
@@ -51,10 +53,13 @@ int StarParticleRadTransfer(LevelHierarchyEntry *LevelArray[], int level,
   /* If sources exist, delete them */
 
   RadiationSourceEntry *dummy;
-  while (GlobalRadiationSources != NULL) {
-    dummy = GlobalRadiationSources;
-    GlobalRadiationSources = GlobalRadiationSources->NextSource;
-    delete dummy;
+  if (GlobalRadiationSources!=NULL){
```

Comment:
I think this is the same block as in `ActiveParticleInitialize.C` Maybe this could be a separate function like `DeleteAllRadiationSources`?

Reply:
Done!